### PR TITLE
[wdspec] Remove uses of navigation with `wait=none`

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Changing the test to not rely on the `browsingContext.navigate`. This makes the test more isolated as it does not rely on the browser module. This also unblocks the `chromium-bidi` implementation of interception as there are issue related to `wait=none`.

### CODE REVIEW

**Comment by juliandescottes** [comment]
@Lightning00Blade can you detail the motivation for this change? 

Internally (especially with @whimboo ), we've been discussing a bit about using `navigate` vs `fetch`, and the conclusion was that using `navigate` in our tests was a good way to improve coverage for this important command. At the same time I agree that using `fetch` will make the tests more isolated, but curious to know if there are other reasons?

**Comment by OrKoN** [comment]
@juliandescottes main reason is to be able to progress on the network interception without needing to resolve all navigation issues. In particular, chromium-bidi would dead-lock itself because it does not really implement with readiness=none correctly: it actually waits for the network request to start before returning, and the test cannot unblock the intercepted navigation request because it also waits for navigate to resolve. @Lightning00Blade please add it to the description if my understanding is correct.

**Review by OrKoN — CHANGES_REQUESTED** [review]
looks like some tests are regressing with this change. PTAL

File: webdriver/tests/bidi/network/auth_required/auth_required.py
Comment (OrKoN) [inline webdriver/tests/bidi/network/auth_required/auth_required.py:29]: Can we alternatively solve it by not blocking on the navigate call? and rely on the event instead

**Comment by juliandescottes** [comment]
> @juliandescottes main reason is to be able to progress on the network interception without needing to resolve all navigation issues.

Makes sense to me, we should probably add a few navigate tests for pages with auth prompts in another PR, to keep covering this use case.

**Review by OrKoN — APPROVED** [review]
LGTM

**Review by OrKoN — APPROVED** [review]

**Comment by Lightning00Blade** [comment]
@juliandescottes can you please take a look? 

For your thoughts about testing navigation request as well, I agree but we should do it in a smarter way. What I mean by that is that we should create a new fixture (something like `create_request`), which should have a toggle for either `fetch` or navigation (this should have a timeout similar to fetch to prevent blocking), we already do that for the `create_block_request` feature. Once that is implementation we can switch all interception test to use it and have the tests parameterized for both cases. This should not block this PR though.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**
